File: src/main/java/org/stt/persistence/stt/STTItemWriter.java
package org.stt.persistence.stt;

import org.stt.model.TimeTrackingItem;
import org.stt.persistence.ItemWriter;

import javax.inject.Inject;
import java.io.PrintWriter;
import java.io.Writer;
import java.util.Objects;

public class STTItemWriter implements ItemWriter {
	private final PrintWriter out;
	private final STTItemConverter converter = new STTItemConverter();

	@Inject
	public STTItemWriter(@STTFile Writer out) {
        this.out = new PrintWriter(Objects.requireNonNull(out));
    }

	@Override
    public void write(TimeTrackingItem item) {
        out.println(converter.timeTrackingItemToLine(item));
    }

	@Override
    public void close() {
        out.close();
    }
}

File: src/main/kotlin/org/stt/persistence/stt/STTItemWriter.kt
package org.stt.persistence.stt

import org.stt.model.TimeTrackingItem
import org.stt.persistence.ItemWriter
import java.io.PrintWriter
import java.io.Writer
import javax.inject.Inject

class STTItemWriter @Inject
constructor(@STTFile out: Writer) : ItemWriter {
    private val out: PrintWriter = PrintWriter(out)
    private val converter = STTItemConverter()

    override fun write(item: TimeTrackingItem) {
        out.println(converter.timeTrackingItemToLine(item))
    }

    override fun close() {
        out.close()
    }
}
